gdal/ogr/ogrct.cpp

```diff
@@ -2061,6 +2073,14 @@ int OGRProjCT::TransformWithErrorCodes(
 /*      Select dynamically the best transformation for the data, if     */
 /*      needed.                                                         */
 /* -------------------------------------------------------------------- */
+    const double dfDefaultTime =
```

Comment:
I don't fully understand this logic -- why do we take the destination epoch and assign it to the input points in some cases?

Reply:
yes, that's a bit tricky, but PROJ can currently only do staticCRS -> dynamicCRS or dynamicCRS -> staticCRS transformations, and in either case, the coordinate epoch of the dynamicCRS must be provided as the input time. Adding that as a comment in the code

Reply:
Ah right -- so ultimately for proper support proj would need support for specific epoch -> specific epoch transforms then? I'm wondering if a user facing warning shouldn't be shown if both source and target have epochs set, since the results would be misleading...

Reply:
> proj would need support for specific epoch -> specific epoch transforms then?

yes, that would also require to have the proj.db / EPSG to incorporate such transforms. The only ones currently are velocity grids for the Canada NAD83(CSRS)v7.  Plate motion by themselves are not really incorporated. If you want to do ATRF2014 @epoch1 to ATRF2014 @epoch2, you can emulate that by going:
ATRF2014 @epoch1 -> GDA2020 and GDA2020 -> ATRF2014 @epoch2 . Basically this is equivalent of using the Helmert transformation of ATRF2014 <--> GDA2020 with a delta_time = epoch2 - epoch1

> I'm wondering if a user facing warning shouldn't be shown if both source and target have epochs set,

will add